### PR TITLE
feat: role based permissions UI

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- adds role creation and organization creation

## Changes

https://user-images.githubusercontent.com/13127476/202569441-844831fc-d2c8-4a2e-b99a-ff96cd893ba5.mov

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## TODO:

- [ ] put behind feature flag
- [ ] Add creation and management for adding role to specific flag
- [ ] Ensure the right people have edit role/org default interface

### CODE REVIEW

**Comment by lharries** [comment]
Looks good! Any chance we can keep the edit button but make it disabled with the tool-tip: "You only have view access to this feature flag". 

That way it's clear why they don't have access and avoids support messages saying they are missing a button

**Comment by lharries** [comment]
Also I'm not sure what "Assigned Edit" means? Is there a better phrasing we can think of this?

**Comment by EDsCODE** [comment]
- [x] add handling for flag based permissions before flag has been initialized
- [x] add information regarding current state of permissions on permission modal
- [x] make feature_flag_role_access call filterable

**Comment by EDsCODE** [comment]
State of UI:

- All UI to support functionality based on backend is implemented!
- I have organization defaults, permissions, and assigning custom roles all under owner or admin only. We should consider making this a little more accessible
- Very little extra messaging and copy around what hierarchies of permissioning exist. It's not particularly clear when a role overrides an organization default
- RE: @lharries's comments +1 to needing better messaging around "assigned edit" and also keeping buttons even if a user doesn't have the access level to perform the action. Right now I've removed most of them

File: latest_migrations.manifest [hunk 1]
```diff
@@ -2,7 +2,7 @@ admin: 0003_logentry_add_action_flag_choices
 auth: 0012_alter_user_first_name_max_length
 axes: 0006_remove_accesslog_trusted
 contenttypes: 0002_remove_content_type_name
-ee: 0014_roles_memberships_access_levels
+ee: 0015_set_organization_role_defaults
```
Comment (macobo) [inline latest_migrations.manifest:5]: Doesn't seem to be committed?

**Comment by annikaschmid** [comment]
I agree with @lharries, in the "view" state, I would either show the buttons as disabled, or have a banner "You have only 'View' access for this feature flag. To make changes, please contact the project owner" or something.

~~BTW who is able to make changes to the feature flag permissions?~~
I found the answer to this. It's organisation admins.

**Comment by annikaschmid** [comment]
Minor: There are a couple of areas where we should change to caps/no caps:
posthog (in text) -> PostHog
Create Role (button) -> Create role
feature flags (in 'resource' column) -> Feature Flags

fine grained -> fine-grained

File: latest_migrations.manifest (hunk 1 above)
Comment (EDsCODE) [reply latest_migrations.manifest:5]: true—this PR wasn't meant to handle any migrations that was leftover from testing i was doing! the base of this PR needs to be updated once ready

**Review by liyiy — APPROVED** [review]
Addressed a couple of the comments above and since this is to get a working UI for internal testing and it's also feature flagged + pay gated, will fix any followups in a new PR, after Thanksgiving :) .

**Pull request merged**